controllers/reconcile_operator.go

```diff
@@ -198,27 +198,30 @@ func (r *OperandRequestReconciler) deleteSubscription(operandName string, reques
 		return err
 	}
 
-	if csv != nil {
-		klog.V(2).Infof("Deleting all the Custom Resources for CSV, Namespace: %s, Name: %s", csv.Namespace, csv.Name)
-		if err := r.deleteAllCustomResource(csv, configInstance, operandName, op.Namespace); err != nil {
-			klog.Error("failed to delete a Custom Resource: ", err)
-			return err
-		}
+	if csv == nil {
```

Comment:
Could we just skip the CSV deleting when the CSV is not found?

For example, in this upgrade pending case https://github.com/operator-framework/operator-lifecycle-manager/issues/1548, the CSV will never be created, which will cause these operators can't be removed.